src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php

```diff
@@ -418,7 +418,7 @@ protected function instantiateObject(array &$data, string $class, array &$contex
 
                     $exception = NotNormalizableValueException::createForUnexpectedDataType(
                         sprintf('Failed to create object because the class misses the "%s" property.', $constructorParameter->name),
-                        $data,
+                        null,
```

Comment:
Shouldn't this be `$parameterData` instead?

Reply:
There is `$parameterData = $data[$key]` statement in above cases. But in this case (with missing parameter) the `$data[$key]` value does not exist and `$parameterData` is empty too.